process.go

```diff
@@ -0,0 +1,263 @@
+package hcsshim
+
+import (
+	"io"
+	"time"
+
+	"encoding/json"
+	"errors"
+	"github.com/Sirupsen/logrus"
+	"syscall"
+)
+
+type hcsProcess struct {
+	handle HcsProcess
+}
+
+type processModifyRequest struct {
+	Operation   string
+	ConsoleSize consoleSize // `json:",omitempty"`
+	//CloseHandle closeHandle `json:",omitempty"`
+}
+
+type consoleSize struct {
+	Height uint16
+	Width  uint16
+}
+
+type closeHandle struct {
+	Handle string
+}
+
+const (
+	stdIn  string = "StdIn"
+	stdOut string = "StdOut"
+	stdErr string = "StdErr"
+)
+
+const (
+	modifyConsoleSize string = "ConsoleSize"
+	modifyCloseHandle string = "CloseHandle"
+)
+
+// Pid returns the process ID of the process within the container.
+func (process *hcsProcess) Pid() (int, error) {
+	title := "HCSShim::Process::Pid"
+	var (
+		processInfo HcsProcessInformation
+		resultp     *uint16
+	)
+
+	if err := hcsGetProcessInfo(process.handle, &processInfo, &resultp); err != nil {
```

Comment:
Finished this before I realized I had access to it in create/open. It will be cached.